tests/agreement.rs

```diff
@@ -0,0 +1,196 @@
+//! Tests of the Binary Byzantine Agreement protocol.
+//!
+//! There are three properties that are tested:
+//!
+//! - Agreement: If any correct node outputs the bit b, then every correct node outputs b.
+//!
+//! - Termination: If all correct nodes receive input, then every correct node outputs a bit.
+//!
+//! - Validity: If any correct node outputs b, then at least one correct node received b as input.
+//!
+//! TODO: Implement adversaries and send BVAL messages at different times.
+
+extern crate hbbft;
+#[macro_use]
+extern crate log;
+extern crate env_logger;
+
+use std::collections::{BTreeMap, VecDeque};
+
+use hbbft::agreement::{Agreement, AgreementMessage};
+
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct NodeId(usize);
+
+/// Message pertaining to a particular Agreement instance.
+type InstanceMessage = (NodeId, AgreementMessage);
+type MessageQueue = VecDeque<InstanceMessage>;
+
+struct TestNode {
+    /// Sender ID.
+    id: NodeId,
+    /// Map from proposer IDs into agreement instances.
+    agreements: BTreeMap<NodeId, Agreement<NodeId>>,
+    /// Queue of triples (sender_id, proposer_id, message).
```

Comment:
Rust code should be in backticks in comments (at least in doc comments with `///`).

Reply:
(Sorry for the spam; I intended to include this with the other comments in the code review.)

Reply:
OK, triples should go once I revert to a single proposer test.